app/Http/Middleware/Locale.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+
+class Locale
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if ($language=$request->session()->get('lang')) {
```

Comment:
```suggestion
        if ($language = $request->session()->get('lang')) {
```
Dùng 1 dấu = ở đây có tác dụng gì e?
Check trường hợp default nếu không có session lang nữa e

Reply:
Dấu = ở đó nó có tác dụng là sẽ **gán** ngôn ngữ đang được lựa chọn lưu ở session **vào** biến $language để thực hiện qua hàm \App::setLocale(**$language**);

Reply:
Trong trường hợp không có session nó sẽ hiện ra văn bản gốc luôn ạ

Reply:
Nếu đơn giản chỉ là gán thì tại sao k viết thẳng là
```
$language = $request->session()->get('lang')
 \App::setLocale($language);

```
Hoặc ngắn hơn là
` \App::setLocale($request->session()->get('lang'));`

Mà lại sử dụng thêm `if` ?

Reply:
Nếu chỉ là **gán** thì anh set luôn như này
```
$language = $request->session()->get('lang')
 \App::setLocale($language);

```
Hoặc ngắn hơn là
` \App::setLocale($request->session()->get('lang'));`

Mà lại sử dụng thêm `if` ?
Theo e có khác biệt gì?

Reply:
dạ vâng em cảm ơn ạ, để em sửa lại

Reply:
Anh đang hỏi mà em
Em sửa lại là nó khác nhau rồi

Reply:
Dạ theo em là nếu gán $language bên trong if thì lúc đó sẽ vừa gán được cho biến và vừa kiểm tra biến đó có null hoặc false gì không. Nếu không có null hoặc false thì mới tiếp tục thực hiện setLocale() ạ